src/contrib/cluster/Akka.Cluster.Sharding.Tests/LeastShardAllocationStrategySpec.cs

```diff
@@ -57,20 +58,35 @@ public void LeastShardAllocationStrategy_should_reallocate_from_region_with_most
 
             // so far regionB has 2 shards and regionC has 0 shards, but the diff is less than rebalanceThreshold
             var r1 = _allocationStrategy.Rebalance(allocations, ImmutableHashSet<string>.Empty).Result;
-            Assert.Equal(r1.Count, 0);
+            r1.Count.Should().Be(0);
 
             allocations = allocations.SetItem(_regionB, new[] { "shard2", "shard3", "shard4" }.ToImmutableList());
             var r2 = _allocationStrategy.Rebalance(allocations, ImmutableHashSet<string>.Empty).Result;
-            Assert.Equal(r2.Count, 1);
-            Assert.Equal(r2.First(), "shard2");
+            r2.Should().BeEquivalentTo(new[] { "shard2", "shard3" });
 
             var r3 = _allocationStrategy.Rebalance(allocations, ImmutableHashSet<string>.Empty.Add("shard4")).Result;
-            Assert.Equal(r3.Count, 0);
+            r3.Count.Should().Be(0);
 
             allocations = allocations.SetItem(_regionA, new[] { "shard1", "shard5", "shard6" }.ToImmutableList());
             var r4 = _allocationStrategy.Rebalance(allocations, ImmutableHashSet<string>.Empty.Add("shard1")).Result;
-            Assert.Equal(r4.Count, 1);
-            Assert.Equal(r2.First(), "shard2");
+            r4.Should().BeEquivalentTo(new[] { "shard2" });
+        }
+
+        [Fact]
+        public void LeastShardAllocationStrategy_should_rebalance_multiple_shards_if_max_simultaneous_rebalances_is_not_exceeded()
+        {
+            var allocations = new Dictionary<IActorRef, IImmutableList<string>>
+            {
+                {_regionA, new []{"shard1"}.ToImmutableList() },
+                {_regionB, new []{ "shard2", "shard3", "shard4", "shard5", "shard6" }.ToImmutableList() },
+                {_regionC, ImmutableList<string>.Empty}
+            }.ToImmutableDictionary();
+
+            var r1 = _allocationStrategy.Rebalance(allocations, ImmutableHashSet<string>.Empty).Result;
+            r1.Should().BeEquivalentTo(new[] { "shard2", "shard3" });
+
+            var r2 = _allocationStrategy.Rebalance(allocations, ImmutableHashSet<string>.Empty.Add("shard2").Add("shard3")).Result;
```

Comment:
You can use simplier construction here `ImmutableHashSet.Create<string>("shard2", "shard3")`